--- ./original/lif1d.c
+++ ./1a/lif1d.c
@@ -49,7 +49,7 @@
 	double		mu;
 	double		s_min;
 	double		s_max;
-	double		*u, *uplus, *sigma, *omega, *omega1;
+	double		*u, *uplus, *sigma, *omega, *omega1, *temp;
 	double		sum;
 	double		time;
 	struct timeval	global_start, global_end, IO_start, IO_end;
@@ -326,8 +326,13 @@
 		/*
 		 * Update network elements and set u[i] = 0 if u[i] > uth
 		 */
+
+		 //CEID 1a
+		 temp = uplus;
+		 uplus = u;
+		 u = temp;
+
 		for (i = 0; i < n; i++) {
-			u[i] = uplus[i];
 			if (u[i] > uth) {
 				u[i] = 0.0;
 				/*
@@ -379,4 +384,3 @@
 
 	return 0;
 }
-
